dimod/constrained.py

```diff
@@ -611,6 +611,21 @@ def fix_variables(self,
 
         return new
 
+    def flip_variable(self, v: Variable):
+        """Flip the specified variable in the objective and constraints.
+
+        Note that this may remove a constraint as being tracked as discrete.
+        Subsequently flipping the variable again will not restore it.
```

Comment:
Should this potential perturbation of the original problem be tracked explicitly when the `flip_variable` method is invoked? Or return something to indicate a constraint was removed? e.g., a var `has_flipped` tracking whether `v` has ever been flipped, so subsequent calls can know a constraint has been removed; or `return True` if constraint was removed.

Reply:
It's a good question. I thought about it and my instinct is no - simply because I don't really expect this case to happen very often. Further, I would expect any user who cares about that level of detail to also be sophisticated enough to track it themselves. Really this is just covering an edge case.

Reply:
OK, makes sense.

Reply:
now how do I add my first "LGTM"...

Reply:
Top right corner of "files changed", there is a "Review changes" button